iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/utils/WALEntryPosition.java

```diff
@@ -135,6 +136,26 @@ public FileChannel openReadFileChannel() throws IOException {
     }
   }
 
+  public WALInputStream openReadFileStream() throws IOException {
+    if (isInSealedFile()) {
+      walFile = walNode.getWALFile(walFileVersionId);
+      return new WALInputStream(walFile);
+    } else {
+      try {
+        walFile = walNode.getWALFile(walFileVersionId);
+        return new WALInputStream(walFile);
+      } catch (IOException e) {
+        // unsealed file may be renamed after sealed, so we should try again
+        if (isInSealedFile()) {
+          walFile = walNode.getWALFile(walFileVersionId);
+          return new WALInputStream(walFile);
+        } else {
+          throw e;
+        }
+      }
+    }
```

Comment:
It seems the "else" branch is just enough, and the branch above is not significantly different from it.

Reply:
Same as above, this is the original design.

Reply:
Just leave a TODO here, then.

Reply:
Done